### PR TITLE
8265185: make sure all code snippets in jdk.incubator.foreign javadoc comments are proper

### PR DESCRIPTION
**Pull request by sundararajana**
checked by compiling code snippets

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265185](https://bugs.openjdk.java.net/browse/JDK-8265185): make sure all code snippets in jdk.incubator.foreign javadoc comments are proper

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/501/head:pull/501` \
`$ git checkout pull/501`

Update a local copy of the PR: \
`$ git checkout pull/501` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/501/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 501`

View PR using the GUI difftool: \
`$ git pr show -t 501`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/501.diff">https://git.openjdk.java.net/panama-foreign/pull/501.diff</a>

</details>

### CODE REVIEW

**Review by mcimadamore — APPROVED** [review]
Looks good (lots of changes :-) ).
Thanks for going through the samples.

File: src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/package-info.java [hunk 1]
```diff
@@ -168,8 +170,9 @@
  * method, so that dereference can happen without the need of creating any additional segment instances:
  *
  * <pre>{@code
+ResourceScope scope = ... // initialize a resource scope object
 MemoryAddress addr = ... //obtain address from native code
-int x = MemoryAccess.getIntAtOffset(MemorySegment.ofNative(), addr.toRawLongValue());
+int x = MemoryAccess.getIntAtOffset(MemorySegment.allocateNative(scope), addr.toRawLongValue());
```
Comment (JornVernee) [inline src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/package-info.java:175]: Shouldn't this use `globalNativeSegment` instead? We don't have an allocateNative overload that just takes a scope.

Comment (sundararajana) [reply src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/package-info.java:175]: missed it! damage done now. may be part of next patch ...

**Pull request closed without merging**